Include Kissat in the Kani bundle

zhassan-aws opened the pull request:
### Description of changes: 

Install Kissat as part of the Kani setup, and include it in the Kani release bundle.

This is required by a PR that adds a `kani::solver` attribute (whic will follow shortly).

**Keeping this as a draft until we integrate a CBMC version that fixes #1962.**

### Resolved issues:

Towards #1656 

### Related RFC:

<!--
Link to the Tracking RFC issue if this work implements part of an RFC.
-->
Optional #ISSUE-NUMBER.

### Call-outs:

<!-- 
Address any potentially confusing code. Is there code added that needs to be cleaned up later? Is there code that is missing because it’s still in development? 
-->

### Testing:

* How is this change tested? A test is included in a PR that will follow shortly.

* Is this a refactor change? No

### Checklist
- [X] Each commit message has a non-empty body, explaining why the change was made
- [X] Methods or procedures are documented
- [X] Regression or unit tests are included, or existing tests cover the modified code
- [X] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

rahulku commented:
do we need to include this in the dependencies also?

zhassan-aws commented:
> do we need to include this in the dependencies also?

Good point. Added.

celinval commented:
Can we also check the expected version as part of the regression script?

For the follow up PR, can you also please add a test that uses kissat as part of the bundle tests.

zhassan-aws commented:
The follow-up PR is #2088. I'll make sure to add a bundle test to it.

zhassan-aws commented:
> Can we also check the expected version as part of the regression script?

Added a check.

> For the follow up PR, can you also please add a test that uses kissat as part of the bundle tests.

I added a test to this PR since #2088 needs some work. The test is invoked using `--enable-unstable --cbmc-args --external-sat-solver kissat`. The test will fail until #1962 is fixed.

celinval commented on scripts/kani-regression.sh:
```diff
@@ -5,6 +5,25 @@
 # To enable "unsound_experimental features, run as follows:
 # `KANI_ENABLE_UNSOUND_EXPERIMENTS=1 scripts/kani-regression.sh`
 
+# Check if kissat has the minimum required version
```
can you move this to its own check_kissat_version? Or even augment the CBMC to include the kissat version.

zhassan-aws replied:
I moved it to a separate script. Didn't feel it should be part of the CBMC one. Perhaps as a cleanup, we can create a single script, e.g. `check_dep_versions` that calls the other scripts.

celinval commented on docs/src/build-from-source.md:
```diff
@@ -30,6 +31,7 @@ git submodule update --init
 ./scripts/setup/ubuntu/install_deps.sh
 ./scripts/setup/ubuntu/install_cbmc.sh
 ./scripts/setup/ubuntu/install_viewer.sh
+./scripts/setup/install_kissat.sh
```
Can we make the `install_deps.sh` to invoke all the other installation scripts? We can keep the other scripts since they are handy to update versions, but this list is starting to grow too much.

zhassan-aws replied:
I'll do this in a follow-up PR.

celinval commented on scripts/setup/install_kissat.sh:
```diff
@@ -0,0 +1,31 @@
+fi
+
+
+# Kissat release
```
Can we check if the version is already installed first?

zhassan-aws replied:
Done.

celinval commented on .github/workflows/kani.yml:
```diff
@@ -156,6 +156,7 @@ jobs:
             docker run -w /tmp/kani/tests/cargo-kani/simple-lib  $tag  cargo kani
```
These should also be a separate script.

zhassan-aws replied:
I'll do this one as well in a follow-up PR.

celinval reviewed (approved):
I suggested a few little improvements to make our lives easier, but not a big deal if you don't do that as part of this PR. But something for us to keep in mind. Thanks!
